pkg/services/auth/github_test.go

```diff
@@ -91,57 +111,86 @@ var _ = Describe("Github Device Flow", func() {
 		Expect(cliOutput.String()).To(ContainSubstring(verificationUri))
 	})
 
-	// These auth flow tests are failing intermittently, so bypass them for now
-	XDescribe("pollAuthStatus", func() {
-		It("retries after a slow_down response from github", func() {
-			rt := newMockRoundTripper(3, token)
-			client.Transport = &testServerTransport{testServeUrl: ts.URL, roundTripper: rt}
-			interval := 5 * time.Second
+	Describe("pollAuthStatus", func() {
+		var rt *mockAuthRoundTripper
+		var s *sleeper
+
+		// pollTimes is a convenience function to convert from a series of polling intervals
+		// to their respective polling timestamps, relative to the sleeper type's starting time
+		pollTimes := func(intervals []time.Duration) []time.Time {
```

Comment:
:question: Why a `func` literal here? I might be missing it but I don't see what we are closing over (if that is the itnent).

Reply:
You’re correct that we’re not closing over anything here. The intent is to keep the helper function scoped to the tests where we’re using it, for test readability as much as anything else.

I’m not sure why Go doesn’t allow `func` declarations within functions to use the same syntax as top-level ones; there’s probably some reason it would make the compiler more complicated.

Reply:
So the argument is "it's only ever used in this one place, so I put it right here"? Makes sense to me.